### PR TITLE
[2.7 CI][New Persistent Mode][5/n] :book: Doc examples for external code

### PR DESCRIPTION
**Pull request by woshiyyya**
## Why are these changes needed?

Enables the new persistence mode for 📖 **Doc examples for external code**.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/external/pytorch_tutorials_hyperparameter_tuning_tutorial.py [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # -*- coding: utf-8 -*-
 # flake8: noqa
+import os
+os.environ["RAY_AIR_NEW_PERSISTENCE_MODE"] = "1"
```
Comment (matthewdeng) [inline doc/external/pytorch_tutorials_hyperparameter_tuning_tutorial.py:4]: I don't think we need this do we?

File: doc/external/pytorch_tutorials_hyperparameter_tuning_tutorial.py [hunk 2]
```diff
@@ -42,15 +45,17 @@
...
 import torch.nn as nn
 import torch.nn.functional as F
 import torch.optim as optim
 from torch.utils.data import random_split
 import torchvision
 import torchvision.transforms as transforms
+import ray
 from ray import train, tune
-from ray.train import Checkpoint
+from ray.train._checkpoint import Checkpoint
```
Comment (matthewdeng) [inline doc/external/pytorch_tutorials_hyperparameter_tuning_tutorial.py:58]: Same for this change, we should still use `ray.train`

File: doc/external/pytorch_tutorials_hyperparameter_tuning_tutorial.py (hunk 1 above)
Comment (woshiyyya) [reply doc/external/pytorch_tutorials_hyperparameter_tuning_tutorial.py]: make sense, removed these

File: doc/external/pytorch_tutorials_hyperparameter_tuning_tutorial.py [hunk 3]
```diff
@@ -42,13 +42,15 @@
...
 import os
+from tempfile import TemporaryDirectory
 import torch
 import torch.nn as nn
 import torch.nn.functional as F
 import torch.optim as optim
 from torch.utils.data import random_split
 import torchvision
 import torchvision.transforms as transforms
+import ray
```
Comment (matthewdeng) [inline doc/external/pytorch_tutorials_hyperparameter_tuning_tutorial.py:53]: Do we need this? Or should this be `import ray.train`

Comment (woshiyyya) [reply doc/external/pytorch_tutorials_hyperparameter_tuning_tutorial.py:53]: yeah, or just use `train` only. Let me remove it

**Review by matthewdeng — APPROVED** [review]

**Pull request merged**